src/west/app/project.py

```diff
@@ -640,6 +645,45 @@ def _die_if_manifest_project_filter(self, action):
                      'the manifest while projects are made inactive by the '
                      'project filter.')
 
+    def _untracked(self):
+        ppaths = []
+        untracked = []
+        for project in self._projects(None):
+            # We do not check for self.manifest.is_active(project) because
+            # inactive projects are still considered "tracked folders"
+            ppaths.append(Path(project.abspath).resolve(strict=False))
+
+        def _find_untracked(folder):
+            self.dbg(f'_find_untracked in: {folder}')
+            for e in [e.resolve() for e in folder.iterdir()]:
+                if e.is_dir():
+                    self.dbg(f'processing folder: {e}')
+                    for ppath in ppaths:
+                        # We cannot use samefile() because it requires the file
+                        # to exist (not the case with inactive projects)
+                        if ppath == e:
+                            # We hit a project root folder, skip it
+                            break
+                        elif e in ppath.parents:
+                            self.dbg(f'recursing into: {e}')
+                            _find_untracked(e)
+                            break
+                    else:
+                        # No match, untracked folder
+                        untracked.append(e)
+                        continue
+                else:
+                    # Untracked file
+                    untracked.append(e)
+
+        # Avoid using Path.walk() since that returns all files and folders under
+        # a particular folder, which is overkill in our case. Instead, recurse
+        # only when required.
+        _find_untracked(Path(self.topdir))
+        self.inf('Untracked files and folders:')
```

Comment:
Does this need to be printed? This would prevent us from using this command in some scriptsl

Reply:
Removed

Reply:
Are we committing to a certain, scriptable format? I don't have a strong opinion but if not then that should be clear in the --help.

Reply:
Good point. I will add it.